iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/ProjectResourceFactory.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.kubernetes.visitors;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.CheckForNull;
+import org.sonar.iac.common.api.tree.HasTextRange;
+import org.sonar.iac.common.checks.PropertyUtils;
+import org.sonar.iac.common.checks.TextUtils;
+import org.sonar.iac.common.checks.Trilean;
+import org.sonar.iac.common.yaml.tree.MappingTree;
+import org.sonar.iac.common.yaml.tree.ScalarTree;
+import org.sonar.iac.common.yaml.tree.SequenceTree;
+import org.sonar.iac.common.yaml.tree.TupleTree;
+import org.sonar.iac.common.yaml.tree.YamlTree;
+import org.sonar.iac.kubernetes.model.LimitRange;
+import org.sonar.iac.kubernetes.model.LimitRangeItem;
+import org.sonar.iac.kubernetes.model.ProjectResource;
+import org.sonar.iac.kubernetes.model.ServiceAccount;
+
+public final class ProjectResourceFactory {
+  private ProjectResourceFactory() {
+    // utility class
+  }
+
+  @CheckForNull
+  public static ProjectResource createServiceAccount(MappingTree tree) {
+    if (!PropertyUtils.valueIs(tree, "kind", kind -> TextUtils.isValue(kind, "ServiceAccount").isTrue())) {
+      return null;
+    }
+
+    var automountServiceAccountTokenTree = PropertyUtils.value(tree, "automountServiceAccountToken")
+      .map(ScalarTree.class::cast);
+    var automountServiceAccountToken = automountServiceAccountTokenTree
+      .map(TextUtils::trileanFromTextTree)
+      .orElse(Trilean.UNKNOWN);
+    var name = PropertyUtils.value(tree, "metadata", MappingTree.class)
+      .flatMap(metadata -> PropertyUtils.value(metadata, "name"))
+      .map(ScalarTree.class::cast)
+      .map(ScalarTree::value);
+
+    return name.map(s -> new ServiceAccount(s, automountServiceAccountToken, automountServiceAccountTokenTree.map(HasTextRange::textRange).orElse(null)))
```

Comment:
What if `ServiceAccount` doesn't have a defined `name`? I guess it is useless as another resource can't reference them. 🤔

Reply:
Yes, then the method would return `null`, and it should be filtered out by the caller.